addons/portal/controllers/portal.py

```diff
@@ -344,44 +428,61 @@ def attachment_remove(self, attachment_id, access_token=None):
 
         return attachment_sudo.unlink()
 
-    def details_form_validate(self, data):
+    @http.route(['/my/main_address_confirmation_warnings'], type='json', auth="user", website=True)
+    def main_address_confirmation_warnings(self, data, **kw):
+        return [_("Are you sure you want to edit your main address?")]
```

Comment:
Is this really necessary to make a controller to return a string ?
You don't add any other warnings in the list using override so.. seems useless.
--> you did it well with delete confirmation

Reply:
Yup, if this is the code about 'accounting b2', would simplify it and we will discuss the complete version afterwards when working on upcoming tasks :)